### PR TITLE
Simplify `JoinToTimeSpineNode`

### PR DESCRIPTION
**Pull request by courtneyholcomb**
A bunch of cleanup in the dataflow to SQL logic for the `JoinToTimeSpineNode`. There was a lot of duplication and hard to read code here.

Notes:
- Please review by commit. The more complex commits have extended commit descriptions to help explain the changes.
- There are many SQL changes here but none of them functional changes. The changes to optimized snapshots might be best to focus on.

### CODE REVIEW

File: metricflow/plan_conversion/dataflow_to_sql.py [hunk 1]
```diff
@@ -322,54 +323,54 @@ def _make_time_spine_data_set(
         """
         time_spine_table_alias = self._next_unique_table_alias()
 
-        agg_time_dimension_specs = {instance.spec for instance in agg_time_dimension_instances}
+        queried_specs = {instance.spec for instance in agg_time_dimension_instances}
```
Comment (courtneyholcomb) [inline metricflow/plan_conversion/dataflow_to_sql.py]: Renamed to differentiate more clearly from `required_specs`.

File: metricflow/plan_conversion/dataflow_to_sql.py [hunk 2]
```diff
@@ -322,54 +323,54 @@ def _make_time_spine_data_set(
...
         specs_required_for_where_constraints = {
             spec
             for constraint in time_spine_where_constraints
             for spec in constraint.linkable_spec_set.time_dimension_specs
         }
-        required_time_spine_specs = sorted(  # sorted for consistency in snapshots
-            agg_time_dimension_specs.union(specs_required_for_where_constraints),
+        required_specs = sorted(  # sorted for consistency in snapshots
```
Comment (courtneyholcomb) [inline metricflow/plan_conversion/dataflow_to_sql.py]: TODO: debug this for date part / maybe custom grain too. Either sort by qualified name or just leave the order as it was sent to you

File: tests_metricflow/snapshots/test_derived_metric_rendering.py/SqlQueryPlan/DuckDB/test_offset_to_grain_metric_filter_and_query_have_different_granularities__plan0_optimized.sql [hunk 3]
```diff
@@ -19,8 +19,8 @@ FROM (
   FROM (
     -- Join to Time Spine Dataset
     SELECT
-      DATE_TRUNC('month', subq_12.ds) AS metric_time__month
-      , subq_10.metric_time__day AS metric_time__day
+      subq_12.ds AS metric_time__day
```
Comment (courtneyholcomb) [inline tests_metricflow/snapshots/test_derived_metric_rendering.py/SqlQueryPlan/DuckDB/test_offset_to_grain_metric_filter_and_query_have_different_granularities__plan0_optimized.sql:22]: Here you can see we are now pulling the join column from the time spine, where in the last PR we were not.

**Review by plypaul — APPROVED** [review]
I've been giving some thought to the complexity around the time spine, and semantic graph could help there. The specifics on how that would be handled are a little murky, but I'll let you know once I get something more concrete there.

**Pull request merged**